src/tests/ftest/container/snapshot_functionality_test.py

```diff
@@ -0,0 +1,539 @@
+            self.fail(
+                "##(2)Error when retrieving the snapshot data: %s"
+                , str(error))
+        self.log.info("==(2)snapshot_list[ind]=%s", snapshot)
+        self.log.info("==snapshot.epoch=  %s", snapshot.epoch)
+        self.log.info("==written thedata=%s", thedata)
+        self.log.info("==thedata2.value= %s", thedata2.value)
+        if thedata2.value != thedata:
+            raise Exception("##(2)The data in the snapshot is not the "
+                            "same as the original data")
+        self.log.info("==Snapshot data matches the data originally "
+                      "written.")
+
+        #(3)Test snapshot with an invalid container handler
+        self.log.info("==(3)Snapshot with an invalid container handler.")
+        if self.invalid_snapshot_test(self.container, epoch):
+            self.log.info("==>Negative test 1, expecting failed on taking "
+                          "snapshot with an invalid container.coh: %s"
+                          , self.container)
+        else:
+            self.fail(
+                "##(3)Negative test 1 passing, expecting failed on"
+                " taking snapshot with an invalid container.coh: %s"
+                , self.container)
+
+        #(4)Test snapshot with a NULL container handler
+        self.log.info("==(4)Snapshot with a NULL container handler.")
+        if self.invalid_snapshot_test(None, epoch):
+            self.log.info("==>Negative test 2, expecting failed on taking "
+                          "snapshot on a NULL container.coh.")
+        else:
+            self.fail("##(4)Negative test 2 passing, expecting failed on "
+                      "taking snapshot with a NULL container.coh.")
+
+        #(5)Test snapshot with an invalid epoch
+        self.log.info("==(5)Snapshot with a NULL epoch.")
+        if self.invalid_snapshot_test(self.container.coh, None):
+            self.log.info("==>Negative test 3, expecting failed on taking "
+                          "snapshot with a NULL epoch.")
+        else:
+            self.fail("##(5)Negative test 3 passing, expecting failed on "
+                      "taking snapshot with a NULL epoch.")
+
+
+    def test_snapshot_info(self):
+        """
+        Test ID: DAOS-1386 Test container SnapShot information
+                 DAOS-1371 Test list snapshots
+                 DAOS-1395 Test snapshot destroy
+                 DAOS-1402 Test creating multiple snapshots
+        Test Description:
+                (1)Create an object, write random data into it, and take
+                   a snapshot.
+                (2)Make changes to the data object. The write_an_obj function
+                   does a commit when the update is complete.
+                (3)Verify the data in the snapshot is the original data.
+                   Get a handle for the snapshot and read the object at dkey,
+                   akey. Compare it to the originally written data.
```

Comment:
Good. You should just merge this with the previous test and do:
1. write data
2. take snapshot
3. read data back & verify content
4. overwrite data
5. read data from snapshot & verify content
...

Reply:
Updated.